src/Common/DateLUTImpl.h

```diff
@@ -641,7 +641,7 @@ class DateLUTImpl
     {
         const LUTIndex i = toLUTIndex(v);
         /// We add 8 to avoid underflow at beginning of unix epoch.
-        return toDayNum(i + 8 - toDayOfWeek(i)) / 7;
+        return toDayNum(i + (8 - toDayOfWeek(i))) / 7;
```

Comment:
`i - toDayOfWeek(i) + 8`?

Reply:
It will  create overflow on left range boundary. 
Consider:
```
$ cal 1 1900
    January 1900      
Su Mo Tu We Th Fr Sa  
    1  2  3  4  5  6  
 7  8  9 10 11 12 13  
14 15 16 17 18 19 20  
21 22 23 24 25 26 27  
28 29 30 31
```
Also:
`select toDayOfWeek(toDate32('1900-01-01'))`
1
So 1900-01-01 is Monday. 

Then your formula `i - toDayOfWeek(i) + 8`:

`1900-01-01 - 1 + 8`

First it makes:
`1900-01-01 - 1 `
Which will be truncated to 1900-01-01 which is wrong.
Then `1900-01-01 + 8 ` will give wrong 1900-01-09

Reply:
It's because `operator+` on `LUTIndex`es is called. Got it.